class.jetpack.php

```diff
@@ -4519,7 +4519,7 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 		}
 
 		if ( $from ) {
-			$url = add_query_arg( 'from', $from, $url );
+			$url = add_query_arg( 'from', sanitize_text_field( $from ), $url );
```

Comment:
Instead of sanitizing here, would it make sense to change the return at the end of the function?

```php
return $raw ? esc_url_raw( $url ) : esc_url( $url );
```

Reply:
This is sanitizing the `from` property, which is a slug like format (see some examples: https://github.com/Automattic/jetpack/pull/12351#discussion_r283720516) and not a URL.

Reply:
Yep. I get that. Let me clarify my question.

Do we need to sanitize the `from` here if we change the return of `build_connect_url` to properly escape the URL?

Currently, the return from `build_connect_url` is:

```php
return $raw ? $url : esc_url( $url );
```

Should it be:

```php
return $raw ? esc_url_raw( $url ) : esc_url( $url );
```

Reply:
That's a good point. I made that change in 71974baa132565cf787a918a4f2ed68f2eb16efc